app/models/post.rb

```diff
@@ -18,6 +18,23 @@ class Post < ApplicationRecord
       Arel.sql("posts.likes_count / EXP(DATE_PART('day', NOW() - posts.created_at) / 4.0) DESC")
     )
   }
+  scope :filtered_by_date, lambda { |date_criteria|
+    where('published_at >= ?', date_criteria)
+  }
+  scope :filtered_by_text, lambda { |text|
+    joins(:user).where(
+      Arel.sql('users.first_name LIKE :search OR users.last_name LIKE :search OR users.nickname LIKE :search
+                OR title LIKE :search OR body LIKE :search'),
+      search: "%#{text}%"
+    )
+  }
+
+  scope :filter_and_sort, lambda { |date_criteria, text_criteria, sort_criteria|
+    posts = self
+    posts = posts.filtered_by_date(date_criteria) if date_criteria.present?
+    posts = posts.filtered_by_text(text_criteria) if text_criteria.present?
+    posts.public_send("ordered_by_#{sort_criteria}")
```

Comment:
The idea of having scopes is being able to chain them together. Let's rewrite this to be `filtered_by_date(date_criteria).filtered_by_text(text_criteria).public_send("ordered_by_#{sort_criteria}")` and let's add the conditions to the other scopes